core/templates/services/navigation.service.ts

```diff
@@ -13,14 +13,33 @@
 // limitations under the License.
 
 /**
- * @fileoverview Factory for navigating the top navigation bar with
+ * @fileoverview Service for navigating the top navigation bar with
  * tab and shift-tab.
  */
-angular.module('oppia').factory('NavigationService', [function() {
-  var navigation = {
-    activeMenuName: '',
-    ACTION_OPEN: 'open',
-    ACTION_CLOSE: 'close',
+
+
+import { Injectable } from '@angular/core';
+import { downgradeInjectable } from '@angular/upgrade/static';
+
+interface KeyboardEvent {
+  keyCode?: number,
+  shiftKey?: boolean,
+  currentTarget?
+}
+
+interface EventToCodes {
+  enter?,
+  tab?,
+  shiftTab?,
+}
+ @Injectable({
+   providedIn: 'root'
+ })
+export class NavigationService {
+  constructor() {}
+    activeMenuName: string;
+    ACTION_OPEN: string = 'open';
+    ACTION_CLOSE: string = 'close';
     KEYBOARD_EVENT_TO_KEY_CODES: {
```

Comment:
```suggestion
    KEYBOARD_EVENT_TO_KEY_CODES = {
```

Reply:
This as well I think, otherwise the value of this.KEYBOARD_EVENT_TO_KEY_CODES will remain undefined in line 90 of the same file, and all other files where it is being used and exported as navigationService.KEYBOARD_EVENT_TO_KEY_CODES (like in the search bar component) as we are just defining its type here and not assigning it a value.

Reply:
yeah fixed that!